### PR TITLE
Fix broken prompt_reset_on_temperature

### PR DESCRIPTION
**Pull request by Purfview**
Fixing: https://github.com/SYSTRAN/faster-whisper/issues/603

Fixes broken prompt_reset_on_temperature functionality.

Broken because `generate_with_fallback()` doesn't return final temperature.

Regression since PR356 -> https://github.com/SYSTRAN/faster-whisper/pull/356

### CODE REVIEW

File: faster_whisper/transcribe.py [hunk 1]
```diff
@@ -731,6 +735,8 @@ def generate_with_fallback(
             decode_result = max(
                 below_cr_threshold_results or all_results, key=lambda x: x[1]
             )
+            # to pass final temperature for prompt_reset_on_temperature
+            WhisperModel.fallbacks_final_temperature = temperature
```
Comment (funboarder13920) [inline faster_whisper/transcribe.py:739]: Handling this with a global variable is not very clean.
Is there an issue with simply replacing decode_result temperature with the accurate temperature ?

Comment (Purfview) [reply faster_whisper/transcribe.py:739]: > Handling this with a global variable is not very clean.

It's not a global, it's local, a class variable. 
To my unprofessional eye this looked cleanest as other ways looked too intrusive.

> Is there an issue with simply replacing decode_result temperature with the accurate temperature ?

I don't know how to replace it, if you know then please make a proper PR of this fix.

Comment (funboarder13920) [reply faster_whisper/transcribe.py:739]: decode_result = (decode_result[0], decode_result[1], 1, decode_result[3])

Comment (Purfview) [reply faster_whisper/transcribe.py]: Thx. Redone it.

**Review by nguyendc-systran — APPROVED** [review]

**Comment by nguyendc-systran** [comment]
Thanks @Purfview for your PR.

**Pull request merged**